src/pages/task/taskTabs/FileTable/utils.ts

```diff
@@ -0,0 +1,23 @@
+import type { GroupedFiles } from "./types";
+
+/**
+ * `filterGroupedFiles` filters the groupedFiles array from the TaskFilesQuery
+ * @param groupedFiles - the groupedFiles array from the TaskFilesQuery
+ * @param search - the search string
+ * @returns - a new array of groupedFiles that contain the search string
+ */
+const filterGroupedFiles = (groupedFiles: GroupedFiles[], search: string) =>
+  groupedFiles.reduce((acc, groupedFile) => {
+    const filteredFiles = groupedFile?.files?.filter((file) =>
+      file.name.toLowerCase().includes(search.toLowerCase())
```

Comment:
I took a crack at this but decided to split it off into another ticket since its out of scope for refactoring the table components. https://jira.mongodb.org/browse/EVG-20889